src/util/GithubRepositoryUtil.js

```diff
@@ -0,0 +1,64 @@
+const REPO_LIST_KEY = 'repoList';
+
+/**
+ * Returns a Promise of the contents of a file or directory in a GitHub repository.
+ * See https://developer.github.com/v3/repos/contents/#get-contents
+ * @param {string} owner Account owner of the repo
+ * @param {string} repo Repo name
+ * @param {string} path The directory or file to retrieve
+ */
+export async function getContents(owner, repo, path) {
+  const normalizedUrl = `https://api.github.com/repos/${owner}/${repo}/contents/${path}`;
+  const response = await fetch(normalizedUrl);
+  return response.json();
+}
+
+export function getRepoList() {
+  const repoListStr = localStorage.getItem(REPO_LIST_KEY);
+  if (repoListStr) {
+    return deserializeRepoList(repoListStr);
+  } else {
+    return undefined;
+  }
+}
+
+export function addToRepoList(owner, repo, path) {
+  const repoMap = {
+    'owner': owner,
+    'repo': repo,
+    'path': path
+  };
+  const repoList = getRepoList();
+  if (repoList === undefined) {
+    localStorage.setItem(REPO_LIST_KEY, JSON.stringify(repoMap));
+  } else {
+    repoList.push(repoMap);
+    localStorage.setItem(REPO_LIST_KEY, serializeRepoList(repoList));
+  }
+  // console.log(getRepoList());
+}
+
+/* Takes an array of repo maps and serializes it into a string */
+// TODO: URLs need to be sanitized
+function serializeRepoList(repoList) {
```

Comment:
This is unnecessary complexity. Assume that repoList is an array (which is implied by its name) and then always serialize an array.

* Empty means no repositories
* 1 element means one repository

You should be able to get rid of `serializeRepoList` and `deserializeRepoList` entirely.